app/models/user.rb

```diff
@@ -99,28 +99,60 @@ def self.find_for_shibboleth_oauth(auth, _signed_in_resource = nil)
     return authentication.user if authentication&.user
   end
 
+  def self.assign_random_password(user)
+    return if user.nil?
+
+    temp_pass = Devise.friendly_token[0, 20] # generate a random token
+    user.password = temp_pass
+    user.password_confirmation = temp_pass
+    user.skip_confirmation!
+    user.save!
+  end
+
   def self.new_with_session(params, session)
     super.tap do |user|
       if (data = session["devise.facebook_data"])
         user.first_name = data["info"]["first_name"]
         user.last_name = data["info"]["last_name"]
         user.email = data["info"]["email"]
-        user.authentications.new(provider: data["provider"],
-                                 uid: data["uid"])
       elsif (data = session["devise.google_oauth2_data"])
         user.first_name = data["info"]["first_name"]
         user.last_name = data["info"]["last_name"]
         user.email = data["info"]["email"]
-        user.authentications.new(provider: data["provider"],
-                                 uid: data["uid"])
       elsif (data = session["devise.shibboleth_data"])
         user.email = data["uid"] # email is uid in our case
-        user.authentications.new(provider: "CMU-Shibboleth",
-                                 uid: data["uid"])
+      else
+        return # early exit if not called from an oauth callback
       end
+
+      User.assign_random_password user
```

Comment:
So does this mean that a user is assigned a random password if they log in using oauth? does this happen once or every single time?